src/dex/hashflow/hashflow.ts

```diff
@@ -657,30 +662,60 @@ export class Hashflow extends SimpleExchange implements IDex<HashflowData> {
               .toFixed(0),
           )
         ) {
+          isFailOnSlippage = true;
           const message = `${this.dexKey}-${this.network}: too much slippage on quote ${side} quoteTokenAmount ${quoteTokenAmount} / destAmount ${destAmount} < ${slippageFactor}`;
+          slippageErrorMessage = message;
           this.logger.warn(message);
-          throw new SlippageCheckError(message);
         }
       } else {
         if (quoteTokenAmount < destAmount) {
+          isFailOnSlippage = true;
           // Won't receive enough assets
           const message = `${this.dexKey}-${this.network}: too much slippage on quote ${side}  quoteTokenAmount ${quoteTokenAmount} < destAmount ${destAmount}`;
+          slippageErrorMessage = message;
           this.logger.warn(message);
-          throw new SlippageCheckError(message);
         } else {
           if (
             baseTokenAmount >
             BigInt(slippageFactor.times(srcAmount.toString()).toFixed(0))
           ) {
+            isFailOnSlippage = true;
             const message = `${this.dexKey}-${
               this.network
             }: too much slippage on quote ${side} baseTokenAmount ${baseTokenAmount} / srcAmount ${srcAmount} > ${slippageFactor.toFixed()}`;
+            slippageErrorMessage = message;
             this.logger.warn(message);
-            throw new SlippageCheckError(message);
           }
         }
       }
 
+      let isTooStrictSlippage = false;
+      if (
+        isFailOnSlippage &&
+        side === SwapSide.SELL &&
+        new BigNumber(1)
+          .minus(slippageFactor)
+          .lt(HASHFLOW_MIN_SLIPPAGE_FACTOR_THRESHOLD_FOR_RESTRICTION)
+      ) {
+        isTooStrictSlippage = true;
+      } else if (
+        isFailOnSlippage &&
+        side === SwapSide.BUY &&
+        slippageFactor
+          .minus(1)
+          .lt(HASHFLOW_MIN_SLIPPAGE_FACTOR_THRESHOLD_FOR_RESTRICTION)
+      ) {
+        isTooStrictSlippage = true;
+      }
+
+      if (isFailOnSlippage && isTooStrictSlippage) {
+        this.logger.warn(
+          `${this.dexKey}-${this.network}: Market Maker ${mm} failed to build transaction on side ${side} with too strict slippage. Skipping restriction`,
+        );
+      } else if (isFailOnSlippage && !isTooStrictSlippage) {
+        throw new SlippageCheckError(slippageErrorMessage);
```

Comment:
I am not sure this is correct. If you fail on slippage, you should throw, but restrict only if slippage is not too much. But this code won't throw if slippage is too strict

Reply:
@Verisana updated